src/zenml/feature_stores/base_feature_store.py

```diff
@@ -0,0 +1,65 @@
+#  Copyright (c) ZenML GmbH 2022. All Rights Reserved.
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at:
+#
+#       https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+#  or implied. See the License for the specific language governing
+#  permissions and limitations under the License.
+
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Dict, List, Union
+
+from feast.feature_service import FeatureService  # type: ignore[import]
+from pandas import DataFrame
+
+from zenml.enums import StackComponentType
+from zenml.stack import StackComponent
+
+
+class BaseFeatureStore(StackComponent, ABC):
+    """Base class for all ZenML feature stores."""
+
+    TYPE: ClassVar[StackComponentType] = StackComponentType.FEATURE_STORE
+    FLAVOR: ClassVar[str]
+
+    @abstractmethod
+    def get_historical_features(
+        self,
+        entity_df: Union[DataFrame, str],
+        features: Union[List[str], FeatureService],
+        full_feature_names: bool = False,
+    ) -> DataFrame:
```

Comment:
I think in the code we follow the convention `import pandas as pd` and then use `pd.DataFrame` everywhere. Might be cool to follow that standard

Reply:
Happy to follow. Is there a reason for that convention?

Reply:
No reason but I think it comes from how pandas is written in most codebases. Just a convention. Perhaps the word DataFrame is too common in libraries like this so they like to append the module to make it explicit

Reply:
Got it. Isn't there a performance hit somehow from importing the entire of Pandas vs just a specific module? I think that was somehow the thought in my mind.

Reply:
I think not in this case, as pandas has their imports sorted properly